src/main/java/com/hubspot/jinjava/objects/serialization/PyishObjectMapper.java

```diff
@@ -40,16 +45,50 @@ public static String getAsUnquotedPyishString(Object val) {
   public static String getAsPyishString(Object val) {
     try {
       return getAsPyishStringOrThrow(val);
-    } catch (JsonProcessingException e) {
+    } catch (IOException e) {
+      if (e instanceof LengthLimitingJsonProcessingException) {
+        if (
+          JinjavaInterpreter
+            .getCurrentMaybe()
+            .map(
+              interpreter -> interpreter.getConfig().getExecutionMode().useEagerParser()
+            )
+            .orElse(false)
+        ) {
+          throw new DeferredValueException(String.format("%s: %s", e.getMessage(), val));
+        } else {
+          throw new OutputTooBigException(
+            ((LengthLimitingJsonProcessingException) e).getMaxSize(),
+            ((LengthLimitingJsonProcessingException) e).getAttemptedSize()
+          );
+        }
+      }
       return Objects.toString(val, "");
     }
   }
 
-  public static String getAsPyishStringOrThrow(Object val)
-    throws JsonProcessingException {
-    String string = PYISH_OBJECT_WRITER.writeValueAsString(val);
-    JinjavaInterpreter.checkOutputSize(string);
-    return string;
+  public static String getAsPyishStringOrThrow(Object val) throws IOException {
+    ObjectWriter objectWriter = PYISH_OBJECT_WRITER;
+    Writer writer;
+    Optional<Long> maxOutputSize = JinjavaInterpreter
+      .getCurrentMaybe()
+      .map(interpreter -> interpreter.getConfig().getMaxOutputSize())
+      .filter(max -> max > 0);
+    if (maxOutputSize.isPresent()) {
+      AtomicInteger remainingLength = new AtomicInteger(
```

Comment:
I thought jinjava interpreters were single threaded?

Reply:
They are, I am using this and modifying the value rather than replacing the value every time we want to update it.
It simplifies what we have to do in the `LengthLimitingWriter` since we can just store the AtomicInteger as an instance variable, rather than a `serializerProvider`
When I was trying another approach, limiting both depth and width of serialization, I created a separate object that held two AtomicIntegers, but since there's just a since AtomicInteger I didn't make a wrapper class.